lib/wallet/src/Cardano/Wallet/Write/Tx.hs

```diff
@@ -491,9 +619,74 @@ utxoFromTxOutsInLatestEra
 utxoFromTxOutsInLatestEra = withStandardCryptoConstraint RecentEraBabbage $
     Shelley.UTxO . Map.fromList . map (second castTxOut)
 
+--------------------------------------------------------------------------------
+-- Tx
+--------------------------------------------------------------------------------
+
+modifyTxOutputs
+    :: RecentEra era
+    -> (TxOut (ShelleyLedgerEra era) -> TxOut (ShelleyLedgerEra era))
+    -> Core.TxBody (ShelleyLedgerEra era)
+    -> Core.TxBody (ShelleyLedgerEra era)
+modifyTxOutputs era f body = case era of
+    RecentEraBabbage -> body
+        { Babbage.outputs = mapSized f <$> Babbage.outputs body
+        }
+    RecentEraAlonzo -> body
+        { Alonzo.outputs = f <$> Alonzo.outputs body
+        }
+  where
+    mapSized f' = mkSized . f' . sizedValue
+
+txBody
+    :: RecentEra era
+    -> Core.Tx (ShelleyLedgerEra era)
+    -> Core.TxBody (ShelleyLedgerEra era)
+txBody RecentEraBabbage = Alonzo.body -- same type for babbage
+txBody RecentEraAlonzo = Alonzo.body
+
+-- Until we have convenient lenses to use
+outputs
+    :: RecentEra era
+    -> Core.TxBody (ShelleyLedgerEra era)
+    -> [TxOut (ShelleyLedgerEra era)]
+outputs RecentEraBabbage = map sizedValue . toList . Babbage.outputs
+outputs RecentEraAlonzo = toList . Alonzo.outputs
+
+-- NOTE: To reduce the need for the caller to deal with @CardanoApiEra
+-- (ShelleyLedgerEra era) ~ era@, we quantify this function over @cardanoEra@
+-- instead of @era@.
+--
+-- TODO [ADP-2353] Move to @cardano-api@ related module
+modifyLedgerBody
+    :: (Core.TxBody (ShelleyLedgerEra cardanoEra)
+        -> Core.TxBody (ShelleyLedgerEra cardanoEra))
+    -> Cardano.Tx cardanoEra
+    -> Cardano.Tx cardanoEra
```

Comment:
Just an idea:
```suggestion
    :: txBody ~ Core.TxBody (ShelleyLedgerEra era)
    => (txBody -> txBody)
    -> (Cardano.Tx era -> Cardano.Tx era)
```

Reply:
Will remove the type application in [ADP-2353](https://input-output.atlassian.net/browse/ADP-2353)